src/status_im/contexts/wallet/wallet_connect/responding_events.cljs

```diff
@@ -96,28 +98,42 @@
                  :wallet-connect-event event
                  :event                :wallet-connect/on-sign-error})
      {:fx [[:dispatch [:wallet-connect/reject-session-request]]
+           [:dispatch
+            [:toasts/upsert
+             {:theme theme
+              :text  (i18n/label :t/wallet-connect-something-went-wrong)
+              :type  :negative}]]
            [:dispatch [:wallet-connect/dismiss-request-modal]]]})))
 
 (rf/reg-event-fx
  :wallet-connect/send-response
- (fn [{:keys [db]} [{:keys [result error]}]]
+ (fn [{:keys [db]} [{:keys [result error on-success on-error]}]]
    (let [{:keys [id topic] :as event} (get-in db [:wallet-connect/current-request :event])
          method                       (wallet-connect-core/get-request-method event)
-         web3-wallet                  (get db :wallet-connect/web3-wallet)]
+         web3-wallet                  (get db :wallet-connect/web3-wallet)
+         theme                        (:theme db)]
      {:fx [[:effects.wallet-connect/respond-session-request
             {:web3-wallet web3-wallet
              :topic       topic
              :id          id
              :result      result
              :error       error
              :on-error    (fn [error]
+                            (when on-error
```

Comment:
These guard checks would be better performed by the provider (effect) because it's a bit less fragile. If the caller needs to know if nils are handled or not, it means the  caller needs to know the implementation of the provider. Since Clojure plays nice usually with nil punning, it's nice too if we can build our internal APIs with that in mind.

Reply:
But here the fx will always get the anon function (parent), as we have more calls below. What would be your suggestion for improving this?

Reply:
Sorry about that, while reading the diff I didn't realize the indentation and the rest of the function calls. You're completely right 👍🏼